src/main/java/org/jenkinsci/plugins/workflow/graphanalysis/ForkScanner.java

```diff
@@ -386,6 +391,7 @@ public Fork(BlockStartNode forkNode) {
     protected void setHeads(@Nonnull Collection<FlowNode> heads) {
         if (heads.size() > 1) {
             parallelBlockStartStack = leastCommonAncestor(new LinkedHashSet<FlowNode>(heads));
+            assert parallelBlockStartStack.size() > 0;
```

Comment:
Pointless AFAICT because the next line would just throw an exception were it empty anyway. (That is the reported bug, right?)